tests/src/app-promotion.test.ts

```diff
@@ -0,0 +1,375 @@
+    // assert:  Alice reserved balance to equal 300
+    // assert:  query appPromotion.staked(Alice) equal [200] /// 500 - 300
+    // assert:  query appPromotion.pendingUnstake(Alice) to equal [300]
+    // assert:  query appPromotion.totalStaked() decreased by 300
+  });
+
+  it('will remove from the "staked" map starting from the oldest entry', async () => {
+    // arrange: Alice balance = 1000
+    // arrange: Alice stakes 100
+    // arrange: Alice stakes 200
+    // arrange: Alice stakes 300
+
+    // assert Alice stake is [100, 200, 300]
+
+
+    // act:     Alice calls appPromotion.unstake(30)
+    // assert:  query appPromotion.staked(Alice) to equal [70, 200, 300] /// Can unstake part of stake
+    // assert:  query appPromotion.pendingUnstake(Alice) to equal [30]
+
+    // act:     Alice calls appPromotion.unstake(170)
+    // assert:  query appPromotion.staked(Alice) to equal [100, 300] /// Can unstake one stake totally and one more partialy
+    // assert:  query appPromotion.pendingUnstake(Alice) to equal [30, 170]
+
+    // act:     Alice calls appPromotion.unstake(400)
+    // assert:  query appPromotion.staked(Alice) to equal [100, 300] /// Can totally unstake 2 stakes in one tx 
+    // assert:  query appPromotion.pendingUnstake(Alice) to equal [30, 170, 400]
+  });
+
+  it('will return balance to "available" state after the period of unstaking is finished, and subtract it from "pendingUnstake"', async () => {
+    // arrange: Alice balance = 1000
+    // arrange: Alice stakes 100
+    // act:     Alice calls appPromotion.unstake(30)
+    // act:     Alice waits for pendingPeriod end (7 days for production but faster for tests)
+
+    // assert:  Alice free balance closeTo ~1000 (minus fee)
+    // assert:  query appPromotion.pendingUnstake(Alice) to equal []
+  });
+
+  it('will throws if unstake amount is more than "staked"', async () => {
+    // arrange: Alice balance = 1000
+    // arrange: Alice stakes 100
+
+    // assert:  Alice calls appPromotion.unstake(200) throws
+    // assert:  query appPromotion.staked(Alice) to equal [100] /// stake amount didnt change 100
+  });
+
+  it('for different accounts in one block is possible', async () => {
+    // arrange: Alice, Bob, Charlie, Dave balance = 1000
+    // arrange: Alice, Bob, Charlie, Dave stakes 100
+
+    // act:     Alice, Bob, Charlie, Dave calls appPromotion.unstake(100)
+
+    // assert:  query appPromotion.pendingUnstake(Alice/Bob,Charlie,Dave) to equal []
+  });
+});
+
+describe.skip('Admin adress', () => {
+  it('can be set by sudo only', async () => {
+    // assert:  Sudo calls appPromotion.setAdminAddress(Alice) /// Sudo successfully sets Alice as admin
+    // assert:  Bob calls appPromotion.setAdminAddress(Bob) throws /// Random account can not set admin
+    // assert:  Alice calls appPromotion.setAdminAddress(Bob) throws /// Admin account can not set admin
+  });
+
+  it('can be Substrate address only', async () => {
```

Comment:
Setting an Ethereum address as an admin does not make sense, but we don't need to make this impossible, if this is how CrossAccountId naturally works. 

Check that consequent setting of ethereum address, and then a valid Substrate address doesn't break things.

Reply:
Changed this test to "can be any valid CrossAccountId"